homeassistant/components/powerplanner/binary_sensor.py

```diff
@@ -0,0 +1,128 @@
+"""Power planner plan sensor."""
+from datetime import datetime, timedelta
+import logging
+from threading import Timer
+
+import aiohttp
+import async_timeout
+from powerplanner import PowerplannerHub
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+from homeassistant.util import slugify
+
+from .const import DOMAIN
+from .entity import PowerPlannerEntityBase
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up PowerPlanner from a config entry."""
+
+    er.async_get(hass)
+    hub: PowerplannerHub = hass.data[DOMAIN][config_entry.entry_id]
+    hub.add_sensor_callback = async_add_entities
+    coordinator = UpdateCoordinator(hass, hub, config_entry)
+    coordinator.set_update_interval(12)
+    await coordinator.async_config_entry_first_refresh()
+
+
+class PPSensorEntity(PowerPlannerEntityBase, CoordinatorEntity, BinarySensorEntity):
```

Comment:
My brain is too childish for this


```suggestion
class PowerPlannerSensorEntity(PowerPlannerEntityBase, CoordinatorEntity, BinarySensorEntity):
```